CRM/Odoosync/Sync/Contribution/Data/LineItem.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+class CRM_Odoosync_Sync_Contribution_Data_LineItem extends CRM_Odoosync_Sync_Contribution_Data {
+
+  /**
+   * Account relationship id
+   * (Income Account is)
+   *
+   * @var int
+   */
+  const INCOME_ACCOUNT_RELATIONSHIP_ID = 1;
+
+  /**
+   * Account relationship id
+   * (Sales Tax Account is)
+   *
+   * @var int
+   */
+  const SALES_ACCOUNT_RELATIONSHIP_ID = 10;
+
+  /**
+   * Gets line item params
+   *
+   * @return array
+   */
+  public function retrieve() {
+    $lineItems = [];
+    $lineItemsData = $this->generateItemsData();
+    foreach ($lineItemsData as $lineItem) {
+      $taxNameList = is_null($lineItem['tax_name']) ? [] : [$lineItem['tax_name']];
+      $lineItems[] = [
+        [
+          'name' => 'tax_name',
+          'type' => 'string',
+          'value' => $taxNameList
+        ],
+        [
+          'name' => 'account_code',
+          'type' => 'int',
+          'value' => $lineItem['account_code']
+        ],
+        [
+          'name' => 'name',
+          'type' => 'string',
+          'value' => $lineItem['label']
+        ],
+        [
+          'name' => 'price_subtotal',
+          'type' => 'double',
+          'value' => $lineItem['total']
+        ],
+        [
+          'name' => 'price_unit',
+          'type' => 'double',
+          'value' => $lineItem['unit_price']
+        ],
+        [
+          'name' => 'product_code',
+          'type' => 'string',
+          'value' => $lineItem['product_code']
+        ],
+        [
+          'name' => 'x_civicrm_id',
+          'type' => 'int',
+          'value' => $lineItem['line_item_id']
+        ],
+        [
+          'name' => 'quantity',
+          'type' => 'double',
+          'value' => $lineItem['quantity']
+        ]
+      ];
+    }
+
+    return $lineItems;
+  }
+
+  /**
+   * Gets line item data
+   *
+   * @return array
+   */
+  private function generateItemsData() {
+    $query = "
+      SELECT 
+        (
+          CASE  
+            WHEN 
+              line_item.entity_table = 'civicrm_participant'
+            THEN 
+              'CVEVT'
+            WHEN 
+              line_item.entity_table = 'civicrm_membership'
+            THEN 
+              'CVMEM'
+            WHEN 
+              line_item.entity_table = 'civicrm_booking_slot'
+              OR line_item.entity_table = 'civicrm_booking_sub_slot'
+              OR line_item.entity_table = 'civicrm_booking_adhoc_charges'
+            THEN 
+              'CVBK'
+            WHEN 
+              line_item.entity_table = 'civicrm_contribution'
+            THEN 
+              'CVCTB'
+            ELSE
+              '' 
+            END
+        ) AS product_code,
```

Comment:
I would calculate the product code value inside a separate class method instead of doing it inside the query, you can call this new method when u loop into the results below and u have to pass the entity_table to it as a parameter.